node/network/pov-distribution/src/tests.rs

```diff
@@ -0,0 +1,913 @@
+
+        assert_matches!(
+            handle.recv().await,
+            AllMessages::NetworkBridge(
+                NetworkBridgeMessage::ReportPeer(peer, rep)
+            ) => {
+                assert_eq!(peer, peer_a);
+                assert_eq!(rep, BENEFIT_FRESH_POV);
+            }
+        );
+
+        assert_matches!(
+            handle.recv().await,
+            AllMessages::NetworkBridge(
+                NetworkBridgeMessage::SendValidationMessage(peers, message)
+            ) => {
+                assert_eq!(peers, vec![peer_b.clone()]);
+                assert_eq!(
+                    message,
+                    send_pov_message(hash_a, pov_hash, pov.clone()),
+                );
+            }
+        );
+
+        assert!(!state.peer_state[&peer_b].awaited[&hash_a].contains(&pov_hash));
+    });
+}
+
+#[test]
+fn peer_completing_request_no_longer_awaiting() {
+    let hash_a: Hash = [0; 32].into();
+
+    let peer_a = PeerId::random();
+
+    let (pov_send, pov_recv) = oneshot::channel();
+
+    let pov = make_pov(vec![1, 2, 3]);
+    let pov_hash = pov.hash();
+
+    let mut state = State {
+        relay_parent_state: {
+            let mut s = HashMap::new();
+            let mut b = BlockBasedState {
+                known: HashMap::new(),
+                fetching: HashMap::new(),
+                n_validators: 10,
+            };
+
+            // pov is being fetched.
+            b.fetching.insert(pov_hash, vec![pov_send]);
+
+            s.insert(hash_a, b);
+            s
+        },
+        peer_state: {
+            let mut s = HashMap::new();
+
+            // peer A is registered as awaiting.
+            s.insert(
+                peer_a.clone(),
+                make_peer_state(vec![(hash_a, vec![pov_hash])]),
+            );
+
+            s
+        },
+        our_view: View(vec![hash_a]),
+        metrics: Default::default(),
+    };
+
+    let pool = sp_core::testing::TaskExecutor::new();
+    let (mut ctx, mut handle) = polkadot_node_subsystem_test_helpers::make_subsystem_context(pool);
+
+    executor::block_on(async move {
+        handle_network_update(
+            &mut state,
+            &mut ctx,
+            NetworkBridgeEvent::PeerMessage(
+                peer_a.clone(),
+                send_pov_message(hash_a, pov_hash, pov.clone()),
+            ).focus().unwrap(),
+        ).await.unwrap();
+
+        assert_eq!(&*pov_recv.await.unwrap(), &pov);
+
+        assert_matches!(
+            handle.recv().await,
+            AllMessages::NetworkBridge(
+                NetworkBridgeMessage::ReportPeer(peer, rep)
+            ) => {
+                assert_eq!(peer, peer_a);
+                assert_eq!(rep, BENEFIT_FRESH_POV);
+            }
+        );
+
+        // We received the PoV from peer A, so we do not consider it awaited by peer A anymore.
+        assert!(!state.peer_state[&peer_a].awaited[&hash_a].contains(&pov_hash));
+    });
```

Comment:
have you considered having a local rustfmt.toml file with `hard_tabs = true`? :)

Reply:
I actually have it in my `polkadot` folder, and it works for existing documents, but not new ones without adding it expliclity to the formatter (yet to be tested, assumption).